mypy/checkexpr.py

```diff
@@ -2421,6 +2421,11 @@ def check_list_multiply(self, e: OpExpr) -> Type:
         e.method_type = method_type
         return result
 
+    def visit_assignment_expr(self, e: AssignmentExpr) -> Type:
+        value = self.accept(e.value)
+        self.chk.check_assignment(e.target, e.value)
```

Comment:
It seems to me this direct call will result in allowing assignments to final variables.

Reply:
Good catch, I'll add `self.chk.check_final`.